erpnext/hr/doctype/employee_checkin/employee_checkin.py

```diff
@@ -27,17 +29,25 @@ def validate_duplicate_log(self):
 
 	def fetch_shift(self):
 		shift_actual_timings = get_actual_start_end_datetime_of_shift(self.employee, get_datetime(self.time), True)
-		if shift_actual_timings[0] and shift_actual_timings[1]:
-			if shift_actual_timings[2].shift_type.determine_check_in_and_check_out == 'Strictly based on Log Type in Employee Checkin' and not self.log_type and not self.skip_auto_attendance:
-				frappe.throw(_('Log Type is required for check-ins falling in the shift: {0}.').format(shift_actual_timings[2].shift_type.name))
-			if not self.attendance:
-				self.shift = shift_actual_timings[2].shift_type.name
-				self.shift_actual_start = shift_actual_timings[0]
-				self.shift_actual_end = shift_actual_timings[1]
-				self.shift_start = shift_actual_timings[2].start_datetime
-				self.shift_end = shift_actual_timings[2].end_datetime
-		else:
-			self.shift = None
+		allow_overtime = False
+		if shift_actual_timings[2]:
+			allow_overtime = frappe.db.get_value("Shift Type", shift_actual_timings[2].shift_type.name, "allow_overtime")
+		if not allow_overtime:
```

Comment:
The conditions can be swapped: `if allow_overtime` and `else`. This condition block is confusing

Reply:
there is buffer time for entrance and exit of employees like Employee can come 60 mins before shift timing. but in overtime, the buffer time doesn't matter because we are paying for overtime itself.  so there is no shift_actual_time.
The reason why `self.shift_actual_end ` and `self.shift_actual_end ` is not set if, overtime is allowed.